src/http/routes.cc

```diff
@@ -94,6 +93,9 @@ future<std::unique_ptr<http::reply> > routes::handle(const sstring& path, std::u
             handler->verify_mandatory_params(*req);
             auto r =  handler->handle(path, std::move(req), std::move(rep));
             return r.handle_exception(_general_handler);
+        } catch (const redirect_exception& _e) {
+            *rep = _e.to_reply();
+            rep->done("json");
```

Comment:
This catch was missing previously. It looks like it fixes a bug - can you explain in the commit message what bug this was? Can you perhaps write a unit test for it?